pallets/loans-ref/src/pricing/external.rs

```diff
@@ -0,0 +1,107 @@
+use cfg_primitives::Moment;
+use cfg_traits::{
+	data::{DataCollection, DataRegistry},
+	ops::{EnsureAddAssign, EnsureDiv, EnsureMul, EnsureSub, EnsureSubAssign},
+};
+use cfg_types::adjustments::Adjustment;
+use codec::{Decode, Encode, MaxEncodedLen};
+use frame_support::{self, ensure, RuntimeDebugNoBound};
+use scale_info::TypeInfo;
+use sp_runtime::{traits::Zero, DispatchError, DispatchResult};
+
+use crate::pallet::{Config, Error, PoolIdOf, PriceOf};
+
+/// External pricing method
+#[derive(Encode, Decode, Clone, PartialEq, Eq, TypeInfo, RuntimeDebugNoBound, MaxEncodedLen)]
+#[scale_info(skip_type_params(T))]
+pub struct ExternalPricing<T: Config> {
+	/// Id of an external price
+	pub price_id: T::PriceId,
+
+	/// Number of items associated to the price id
+	pub quantity: T::Balance,
```

Comment:
I would still suggest renaming this. Since this is not the number of items associated, that's the `outstanding_quantity` now, rather it's the equivalent of the `max_borrow_amount` from internal pricing.
```suggestion
	/// Maximum number of items associated to the price id
	pub max_borrow_quantity: T::Balance,
```

Reply:
Ahh, I finally see what you mean by that name. Ok, I agree!